### ISSUE TITLE
Error processing image with Gamma

### ISSUE THREAD

**Issue by Pipozor** [JimBobSquarePants/ImageProcessor#267]
I'm doing on jpeg file:
var image = imageFactory.Load(inStream);
 it returns an exception "Error processing image with Gamma"
The jpeg I'm using seems to have a gamma feature.
Is there any way to avoid this exception or is this will be supported in the future ?
Thanks

**Comment by JimBobSquarePants** [comment]
Hi Pipozar,

I'm going t need a little more information than this. 

- What version of ImageProcessor are you using?
- What size is the image?
- Are you on a 32 bit or 64 bit machine?

The error is being thrown during the conversion from and sRGBA to linear color space as you are most likely running out of contiguous memory.

**Comment by Pipozor** [comment]
Hi JimBob,

Thank your prompt answer.
The problematic image is that sample : https://upload.wikimedia.org/wikipedia/commons/a/a9/Le_Pen%2C_Marine-9586.jpg

* I'm using the last (current) version of ImageProcessor
* I'm on a powerfull 64 bit server machine.
* Exception occurs in the .Load 

```
 var image = imageFactory.Load(inStream);
```

Here is my full function if it can helps:

```
public static bool ImageCropAndResize(string imageSrc, string imageDst, double zoom, double offsetX, double offsetY)
        {
            byte[] photoBytes = File.ReadAllBytes(imageSrc);
            int quality = 90;
            var format = new JpegFormat();

            if (zoom > 1.0)
                zoom = 1.0;

            using (var inStream = new MemoryStream(photoBytes))
            {
                using (var outStream = new MemoryStream())
                {
                    // Initialize the ImageFactory using the overload to preserve EXIF metadata.
                    using (var imageFactory = new ImageFactory(preserveExifData: true))
                    {
                        try
                        {
                            var image = imageFactory.Load(inStream);
                            var newW = (int)(image.Image.Width*zoom);
                            var newH = (int)(image.Image.Height*zoom);
                            var cl = new CropLayer((float)Math.Abs(offsetX), (float)Math.Abs(offsetY), 256, 256, CropMode.Pixels);
                            image.Resize(new Size(newW, newH));
                            image.BackgroundColor(Color.White);

                            image.Crop(cl)
                                .Format(format)
                                .Quality(quality)
                                .Save(outStream);
                        }
                        catch (Exception ex)
                        {
                            FL.Error("ImageCrop => "+ex.Message);
                        }

                        // Save to target file
                        using (FileStream file = new FileStream(imageDst, FileMode.Create, FileAccess.Write))
                        {
                            byte[] bytes = new byte[outStream.Length];
                            outStream.Read(bytes, 0, (int)outStream.Length);
                            file.Write(bytes, 0, bytes.Length);
                            outStream.Close();
                        }
                    }
                }
            }

            return true;
        }
    }
```

**Comment by Pipozor** [comment]
Sorry to have bother you, the problem does not occurs anymore in 2.3.1.
so it's clear

**Comment by JimBobSquarePants** [comment]
Thanks, for the extra code sample. Yeah, there was a bug in that version which caused more memory usage than it should have which meant some larger images would fail. I fixed it in [v2.2.4](https://www.nuget.org/packages/ImageProcessor/2.2.4)

Glad you have it sorted now. :smile:

**Comment by Pipozor** [comment]
Thank you very much for this library. I Love it. Easy to use and a great potential...
All the best.
